File: Java-Programs/Array/SecondLargest.java
//Find Second Largest Number in Array Without Sorting

public class SecondLargest {
	public static void main(String[] args) {
		int arr[] = { 12, 43, 13, 54, 11 };
		int max = arr[0];
		int sec_max = arr[1];
		for (int i = 0; i < arr.length; i++) {
			if (arr[i] > max) {
				sec_max = max;
				max = arr[i];
			} else if (arr[i] > sec_max && arr[i] != max) {
				sec_max = arr[i];
			}
		}
		System.out.println("Second Largest Element Without Sorting:- " + sec_max);
	}
}

File: PHP-Programs/Array/SecondLargest.php

<?php
//Find Second Largest Number in Array Without Sorting
$arr = [12, 43, 13, 54, 11];
$max = $arr[0];
$sec_max = $arr[1];

foreach ($arr as $element) {
    if ($element > $max) {
        $sec_max = $max;
        $max = $element;
    } elseif ($element > $sec_max && $element != $max) {
        $sec_max = $element;
    }
}

echo "Second Largest Element Without Sorting: $sec_max\n";
?>
